source/validate_id.cpp

```diff
@@ -417,7 +417,7 @@ bool idUsage::isValid<SpvOpTypeArray>(const spv_instruction_t* inst,
   } else if (5 == constInst->words.size()) {
     uint64_t value =
         constInst->words[3] | ((uint64_t)constInst->words[4]) << 32;
-    bool signedness = constResultType->second.inst->words[3];
+    bool signedness = constResultType->second.inst->words[3] ? true : false;
```

Comment:
Oh, the implicit conversion still happens -- the `?` takes a `bool` for its first argument. :)

I find the readability argument less than convincing, given that this implicit conversion is a well-understood language feature dating back to the 80s.

I don't have a problem with accepting this particular change (especially since I agree with the other parts), but I would have a problem with a general rule requiring us to always write the silly `? true : false` everywhere just to suppress a single compiler's warning.